src/main/java/com/uid2/shared/auth/OperatorKey.java

```diff
@@ -77,6 +96,7 @@ public boolean equals(Object o) {
                 && this.name.equals(b.name)
                 && this.contact.equals(b.contact)
                 && this.protocol.equals(b.protocol)
+                && this.roles.equals(this.roles)
```

Comment:
@gmsdelmundo-ttd you didn't check for siteId here previously - is there any reason not to?

Reply:
I saw that the `disabled` flag was also not included, so I initially didn't add it. On second thought now, it should be added as well. @sunnywu Should we add both `disabled` and `siteId` to the `equals()` comparison?

Btw @jingyi-gao-ttd If you change the `equals()` comparison, you also have to update `hashCode()` with the same fields as well.

Reply:
@gmsdelmundo-ttd i think it depends on what we use the equals()/hashCode() for. So say if we want to confirm every field is equal then sure we should add both `disabled` and `siteId` in the equals comparison. But what about using the methods in a Set where we want to check if an OperatorKey object is in it but not necessarily caring the attribute like disabled/roles?

Reply:
by just comparing the key/name/contact which has to be unique combo? 

but i tend to agree we should add all fields to the equals comparison and the Set usage is an edge case ?

Reply:
As per our discussion, let's just include all fields in hashCode and equals here

Reply:
I've added disabled, siteId and roles in hashCode and equals